mitmproxy-rs/src/dns_resolver.rs

```diff
@@ -0,0 +1,19 @@
+use hickory_resolver::config::{LookupIpStrategy, ResolverConfig, ResolverOpts};
+use hickory_resolver::TokioAsyncResolver;
+use pyo3::prelude::{pyfunction, PyResult, Python};
+use pyo3::types::PyAny;
+
+#[pyfunction]
+#[pyo3(signature = (host, is_ipv6))]
+pub fn getaddrinfo(py: Python<'_>, host: String, is_ipv6: bool) -> PyResult<&PyAny> {
+    pyo3_asyncio::tokio::future_into_py(py, async move {
+        let mut opts = ResolverOpts::default();
+        if is_ipv6 {
+            opts.ip_strategy = LookupIpStrategy::Ipv6Only;
+        }
+        let resolver = TokioAsyncResolver::tokio(ResolverConfig::default(), opts);
```

Comment:
We should use
https://docs.rs/hickory-resolver/latest/hickory_resolver/system_conf/fn.read_system_conf.html
and then set
https://docs.rs/hickory-resolver/latest/hickory_resolver/config/struct.ResolverOpts.html#structfield.use_hosts_file.
The default config uses Google's DNS servers, which I think some mitmproxy users will not appreciate. :)

Reply:
Both these options are available for unix only, wouldn't that create a problem for windows? And should we make these options user-configurable?

Reply:
See my other comment, they should work on Windows :)